### ISSUE TITLE
Automatic detection of binary classification tasks

### ISSUE THREAD

**Issue by larskotthoff** [mlr-org/mlr#567]
Currently creating a classification task with exactly two classes automatically assumes that one of them is the "positive" class. This has implications for how many other methods behave, e.g. `getPredictionProbabilities()`. I find this behaviour confusing, as completely different objects are returned when a task is deemed binary. There are many cases where a task isn't binary in the sense that there's a positive class, there just happen to be 2 classes.

Can we require the user to explicitly specify that a task is indeed binary (e.g. by specifying the positive class), or have an option to warn when a binary task is detected without an explicitly specified positive class?

**Comment by berndbischl** [comment]
I really dont get what you are saying. Pls provide some examples. If a task has 2 classes, it is binary. Period. And one class has to be class +1.

**Comment by larskotthoff** [comment]
If I remove one class from iris, does one class become positive?

My main point is really that functions behave differently depending on something that's done implicitly without warning. For example, if I write generic code to work with class probabilities, I need to check and handle the case where there are two classes separately because only one vector of probabilities is returned (even though they don't necessarily add up to 1 with the other class).

**Comment by berndbischl** [comment]
Isn't your main point that getProbabilities "drops a dim"? Can you give any other example?
So its about that function and not the task as such?
Its debatable whether doing that in getProbabilities is a good choice.
The reason is that returning a matrix(p, 1-p) is really redundant - and in many cases you immediately throw away the 2nd col.

> I need to check and handle the case where there are two classes separately

No? Just always set "classes" in getProbabilities to the classes of the task? Thats why we have that option....

**Comment by larskotthoff** [comment]
Yes, it's mostly about the function, although whether a task is binary or not probably also has effects in other parts of the code. It's not always the case that the returned matrix is p and 1-p -- the probabilities don't necessarily add up to 1.

And yes, I can set the classes I want probabilities for explicitly, but this requires an extra step on my side.

**Comment by berndbischl** [comment]
> the probabilities don't necessarily add up to 1.

Yes they do! Maths tells us this, there is no other chance?

**Comment by berndbischl** [comment]
> And yes, I can set the classes I want probabilities for explicitly, but this requires an extra step on 
> my side.

What am I supposed to answer here?
